keps/sig-node/4622-topologymanager-max-allowable-numa-nodes/README.md

```diff
@@ -0,0 +1,789 @@
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+No.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+Examine the kubelet configuration of a node to verify the existence of the feature gate and the utilization of the new policy option.
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [ ] Events
+  - Event Reason: 
+- [ ] API .status
+  - Condition name: 
+  - Other field: 
+- [x] Other (treat as last resort)
+  - Details: Inspect the kubelet configuration of the nodes: check feature gate and usage of the new option.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+<!--
+This is your opportunity to define what "normal" quality of service looks like
+for a feature.
+
+It's impossible to provide comprehensive guidance, but at the very
+high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99.9% of /health requests per day finish with 200 code
+
+These goals will help you determine what you need to measure (SLIs) in the next
+question.
+-->
+
+It won't cause any performance regression. So we don't need to define any SLOs for this feature.
+
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+<!--
+Pick one more of these and delete the rest.
+-->
+
+- [ ] Metrics
+  - Metric name:
+  - [Optional] Aggregation method:
+  - Components exposing the metric:
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+<!--
+Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
+implementation difficulties, etc.).
+-->
+
+N/A
```

Comment:
perhaps we can abuse metrics to report the configured value? let's hear from other reviewers

Reply:
/cc @klueska

Reply:
Something to provide a better answer to the "How can an operator determine if the feature is in use by workloads?" might be appropriate here.

Reply:
ok, so this seems to circle back to https://github.com/kubernetes/enhancements/pull/4624/files#r1635931602 / https://github.com/kubernetes/enhancements/pull/4624/files#r1598220901
In addition, may be worth adding explicitly in the text that this is a node feature, so all the pods which qualify to be aligned by topology manager will use implicitly this feature

Reply:
The feature is not used by workloads in any way shape or form. This feature only (potentially) impacts how long it takes for the kubelet to start a workload. If you really want a metric to be emitted to show the value of this setting, I won't block that, but it seems silly to abuse metrics in this way when it's just as easy to check the configuration to see if it's enabled.

Reply:
For the sake of clarity: I'm completely neutral about adding metrics. I tend to agree that adding metrics seems not a good fit in this case but I'm pretty green about the monitoring world, so my intuition may be out of line. So I'm not pushing for metrics and I agree a good example about how to check the config is sufficient.

Reply:
Agree, we should provide an example to check the config.